### PR TITLE
[FEAT] Adds commenting to the Frontend components

### LINKED ISSUE #308: [FEAT] Add commenting to the frontend folder in the codebase
## Feature Description: 
The frontend folder in the codebase currently is missing comments on some files that may be necessary for understanding.
Sections of code that may be difficult to understand should be commented.
Furthermore, remaining empty comments(used to pass ESLint) that were left by the previous Yumble group, need to be filled out.

The changes will be made in frontend/components in the codebase

### PR DESCRIPTION
**Pull request by Dhors**
The frontend code has been supplemented with comments that detail the main components as well as the 7 main screens.
Comments have also been added for logic in some cases that may not be obvious from the code itself.

The empty comments left by the previous Yumble group have also been filled out to give a holistic view of the file and hence the screen.
---

Closes #308

### CODE REVIEW

**Review by absternator — APPROVED** [review]
looks good to me !!!!!!

File: frontend/src/components/3.Preferences/AutocompleteSearchBox.js [hunk 1]
```diff
@@ -41,7 +50,7 @@ const AutocompleteSearchBox = ({ setLocation, sendCoordinates }) => {
             <input
               className={'SearchBox'}
               {...getInputProps({
-                placeholder: 'Grafton, Auckland, New Zealand',
+                placeholder:' Street, City, Country',
```
Comment (reeved) [inline frontend/src/components/3.Preferences/AutocompleteSearchBox.js]: Could you please remove the space before Street. My latest changes now adds padding to the left of the input text :)

File: frontend/src/components/7.Result/ResultPage.js [hunk 2]
```diff
@@ -64,6 +72,9 @@ function ResultPage(props) {
       });
   }, []);
 
+  // Creating the pie chart from the vote data.
+  // Each selected restaurant is shown as a legend and the overal voting
```
Comment (reeved) [inline frontend/src/components/7.Result/ResultPage.js]: small nitpick but it should be 'overall'

**Review by reeved — CHANGES_REQUESTED** [review]
Looking great Daniel! Just some very minor changes requested :)

File: frontend/src/components/3.Preferences/AutocompleteSearchBox.js (hunk 1 above)
Comment (Dhors) [reply frontend/src/components/3.Preferences/AutocompleteSearchBox.js]: Ok changed, thanks for that Reeve

File: frontend/src/components/7.Result/ResultPage.js (hunk 2 above)
Comment (Dhors) [reply frontend/src/components/7.Result/ResultPage.js]: Thanks Revve, my bad I didnt catch that on the spell checking.

**Review by reeved — APPROVED** [review]

**Pull request merged**